bitwarden/clients
# Desktop Modal Experiment

## Tracking

https://bitwarden.atlassian.net/browse/PM-14223

## 📔 Objective

This PR aims to add a way for the desktop window to be presented in a smaller modal mode to enhance the UX for certain flows, like passkeys and SSH Agent.

Todo - but could just as well be separate PRs (are tracked in JIRA as subtasks)

* QA: Test more on windows
* Improvement: Add DeepLinkGuard to make sure the unlock screen forwards you to the specific route
* Improvement: Reset route on windows close (the UI/route opened in the modal is often contextful and should not remain open if the modal is closed)



## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines



- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes


Co-authored-by: @justindbaur
